### PR TITLE
Fix/table alignment

### LINKED ISSUE #11: CI table has alignment issues
This has been an ongoing issue, that has been "patched", but has had no real solution. A lot of variables impact the alignment, including the number of lines entered in the `x_lab` **and** the `CI_label`. The most recent patch addressed multiple lines for the `CI_label`, but did not take the `x_lab` argument into account.

### Example with 2 lines in `CI_label` and 2 lines for `x_lab`:
image

### Example with 2 lines in `CI_label` and 1 line for `x_lab`:
(Notice how the table does not line up with the graph. The misalignment gets worse as you move down the y-axis.)
image

I can further expand the alignment function to take this into account. However the code is already not ideal, and it might be a good idea to completely refactor that instead of patching it. Also probably worth noting that a patch would not be that simple in this case.

### PR DESCRIPTION
**Pull request by barrettk**
Closes #11

### CODE REVIEW

**Comment by barrettk** [comment]
New plot looks much nicer:
image

This PR is still WIP, as the alignment fix ruined the last PR I made that addressed multi-lined table headers

**Comment by barrettk** [comment]
Multiple lines can now be entered for `CI_label`, without messing up the alignment:
image

**Review by seth127 — CHANGES_REQUESTED** [review]
This all looks good, just a few small changes (nothing to the code).

I do want to add tests for this from https://github.com/metrumresearchgroup/pmforest/issues/11#issuecomment-1054500429
> Another key test will be when you change the size of the plot

but I think that's going to be non-trivial. Let's move ahead with this and I might start another branch with that test if I can figure out how.

File: NEWS.md [hunk 1]
```diff
@@ -1,4 +1,21 @@
 
+# pmforest 0.0.2
```
[the remark this answers is not in the document]
Comment (barrettk) [reply NEWS.md:2]: @seth127 bbr has version numbers though. Do we just update the version at the _next_ release?

[the remark this answers is not in the document]
Comment (barrettk) [reply NEWS.md:2]: Misunderstood at first. Will change

File: NEWS.md [hunk 2]
```diff
@@ -1,4 +1,21 @@
...
+
+New Features:
+
+* `shapes` argument. Will denote the shape of the mean/median value. Users can specify one of the following: "square", "diamond", "circle", or "triangle".
```
[the remark this answers is not in the document]
Comment (barrettk) [reply NEWS.md:6]: Done

File: R/forest-constructor.R [hunk 3]
```diff
@@ -257,19 +256,21 @@ forest_constructor <- function(data,
...
-    p2 <- gridExtra::arrangeGrob(p, table_CI, layout_matrix = layout_matrix)
+
+    p2 <- p + table_CI + patchwork::plot_layout(widths = c(plot_width, 12-plot_width))
   } else {
     table_CI <- NULL
-    p2 <- gridExtra::arrangeGrob(p)
+    p2 <- p
   }
 
   # copied this from ggpubr::as_ggplot() so to not need ggpubr (and specifically nloptr)
```
[the remark this answers is not in the document]
Comment (barrettk) [reply R/forest-constructor.R:270]: The part you quoted is not longer in the code, but ill remove mentions of cowplot

File: R/table-plot.R [hunk 4]
```diff
@@ -60,53 +60,30 @@ table_plot <-
...
-        aes(x = x, y = y, label = value),
-        size = text_size,
-        hjust = 0,
         vjust = 0
       ) +
       coord_cartesian(xlim = x_limit,
                       ylim = y_limit,
                       expand = F) +
-      geom_hline(yintercept = lab_title$y - 0.3) + #max(plotdata$ID) + 2.1) +
+      geom_hline(yintercept = y_lines - 0.3) + #max(plotdata$ID) + 2.1) +
```
[the remark this answers is not in the document]
Comment (barrettk) [reply R/table-plot.R:86]: This was addressed in a later commit

File: R/table-plot.R [hunk 5]
```diff
@@ -60,53 +60,30 @@ table_plot <-
...
-        value = tbl_titles
-      )
+    lab <- lab %>% arrange(desc(y))
+    lab <- lab %>% mutate(
+      diff = c(NA, apply(lab[-c(2:3)] , 2 , diff ))
+    )
 
-    # Add extra space to y_limit depending on number of lines in table title
-    num_lines <- stringr::str_count(lab_title$value, "\n") + 1
+    # # Add extra space to y_limit depending on number of lines in table title
```
[the remark this answers is not in the document]
Comment (barrettk) [reply R/table-plot.R:68]: Yeah - I misunderstood the problem multiple times, but once the issue became clear it was much easier to simplify the existing code, and a complete refactor wasnt necessary

**Review by seth127 — APPROVED** [review]
Looks good. After you merge this, check out #15 for me.

**Pull request merged**